contracts/event/events.go

```diff
@@ -1,40 +1,75 @@
 package event
 
-type Instance interface {
-	// Register event listeners to the application.
-	Register(map[Event][]Listener)
-	// Job create a new event task.
-	Job(event Event, args []Arg) Task
-	// GetEvents gets all registered events.
-	GetEvents() map[Event][]Listener
+type Arg struct {
+	Type  string
+	Value any
 }
 
 type Event interface {
 	// Handle the event.
 	Handle(args []Arg) ([]Arg, error)
 }
 
-type Listener interface {
-	// Signature returns the unique identifier for the listener.
-	Signature() string
+type EventListener interface {
+	// Handle the event.
+	Handle(event any, args ...any) error
+}
+
+type EventQueueListener interface {
```

Comment:
The `Event` prefix is duplicated with the package name.

```suggestion
type QueueListener interface {
```